### PR TITLE
feat: v0.3.0 version for java

### PR DESCRIPTION
**Pull request by tzssangglass**
1. add 0.3.0 version for java
2. add maven wrapper to unified maven build version
3. upgrade flatbuffer-java to 2.0.0

Signed-off-by: tzssangglass <tzssangglass@gmail.com>

### CODE REVIEW

**Comment by tzssangglass** [comment]
Just a moment merge, I need to make sure something is working first.

File: mvnw [hunk 1]
```diff
@@ -0,0 +1,310 @@
+#!/bin/sh
```
Comment (spacewander) [inline mvnw:1]: Why mvn wrapper is in need?

Comment (tzssangglass) [reply mvnw:1]: Others can build jars based on this java code with maven without the need to build the maven environment, and shield them from build differences caused by inconsistent maven versions.

**Comment by tzssangglass** [comment]
Merge later and I'll do some verification first.

**Comment by tzssangglass** [comment]
Validation passed.

@spacewander do you think that mvn wrapper is unnecessary? I can remove it.

**Review by spacewander — APPROVED** [review]
The wrapper is acceptable.

**Pull request merged**